src/util/construct-graph.ts

```diff
@@ -0,0 +1,120 @@
+import { Dependency } from "@bc-cr/domain/dependencies";
+import { Graph, Visited } from "@bc-cr/domain/graph";
+
+export function constructGraph(
+  dependencies: Dependency[],
+) {
+  /**
+   * restructure dependencies into a map (directed adjacency list graph):
+   * key = project name
+   * value = array of project names which it depends on
+   */ 
+  const initializedGraph = dependencies.reduce((ig: Graph, current) => {
```

Comment:
please change `ig` to `initialGraph` or anything more descriptive, or change it by `previous` or `prev/curr` since it is inside a `reduce` block